rrdd/rrdd_server.ml

```diff
@@ -274,31 +274,57 @@ let forget_host_ds (ds_name : string) : unit =
     )
 
 let query_possible_dss rrdi =
+  (* We have data sources coming from different places, so we want the union of these;
+     this happens to be the union of the rrdi.rrd (live) and rrdi.rrd.rdd_dss (other).
+     This is straighforward, but those two sets of rrds have different types,
+     hence we need to coerce them into a common type (this is why we have
+     'description not available', and 'units unknown' etc.).
+     Deciding whether a ds is enabled is also not obvious. If we have a 'live'
+     ds, then it is enabled if it exists in the set rrdi.rrd. If we have an 'other'
+     ds, then it is enabled if it is also an enabled 'live' ds, otherwise it is
+     disabled.
+   *)
   let module SMap = Map.Make(String) in
-  let enabled_sources = rrdi.dss
-  |> List.to_seq
-  |> Seq.map (fun ds -> ds.Ds.ds_name, ds)
-  |> SMap.of_seq
+  let open Ds in
+  let open Data_source in
+
+  let live_sources = rrdi.dss in
+  let other_sources = rrdi.rrd.rrd_dss in
+
+  (* we identify dss by their names. assume names which differ only by whitespace or case are actually the same *)
+  let normalize str = str |> Astring.String.trim |> Astring.String.Ascii.lowercase in
```

Comment:
@psafont i found that the names of the data sources are not necessarily equal when they ought to be, but 'normalizing' the strings seems to fix this. this might be a contributing factor as to why f5169c9 wasn't quite right

Reply:
however since the ds names do not match exactly, and here i pick the name from the `rrdi.dss`, this may break other code (although in reality it shouldn't, since previously (f5169c9) the code used ds names from `rrdi.dss` as well)

Reply:
Could you give examples? I also might be good to know what is the code that converts the names of the datasources.

Reply:
i will. it might be that a `trim` is sufficient (i would hope this is the case)

Reply:
turns out that the output of `xe host-data-source-list` changes over time (obviously!). i happened to be looking at the output at the wrong times which led me to believe that `normalize` was making a difference when it wasn't actually. i have removed it as it's not necessary